d3ploy/NOInst.py

```diff
@@ -0,0 +1,265 @@
+        ----------
+        agent : cyclus agent
+            This is the agent that is making the call to the listener.
+        time : int
+            Timestep that the call is made.
+        value : object
+            This is the value of the object being recorded in the time
+            series.
+        """        
+        if time in self.commodity_demand:
+            self.commodity_demand[time] += value
+        else:
+            self.commodity_demand[time] = value
+
+    
+    def demand_calc(self, time):
+        """
+        Calculate the electrical demand at a given timestep (time). 
+        
+        Parameters
+        ----------
+        time : int
+            The timestep that the demand will be calculated at. 
+        Returns
+        -------
+        demand : The calculated demand at a given timestep.
+        """
+        timestep = self.context.dt
+        time = time * timestep
+        demand = self.initial_demand * ((1.0+self.growth_rate)**(time/3.154e+7))
+        return demand
+
+
+    def moving_avg_sup(self, ts, order=1):
+        """
+        Calculates the moving average of a previous [order] entries in
+        timeseries [ts]. It will automatically reduce the order if the
+        length of ts is shorter than the order. 
+
+        Parameters:
+        -----------
+        ts : Array of doubles
+            An array of time series data to be used for the arma prediction
+        order : int
+            The number of values used for the moving average. 
+        Returns
+        -------
+        x : The moving average calculated by the function.         
+        """
+        supply = np.array(list(ts.values()))
+        if order >= len(supply):
+            order = len(supply) * -1
+        else:
+            order *= -1
+        x = np.average(supply[order:])
+        return x
+
+    def predict_arma(self, ts, time=1):
+        """
+        Predict the value of supply or demand at a given time step using the 
+        currently available time series data. This method impliments an ARMA
+        calculation to perform the prediciton. 
+
+        Parameters:
+        -----------
+        ts : Array of doubles
+            An array of time series data to be used for the arma prediction
+        time: int
+            The number of timesteps to predict forward. 
+        Returns:
+        --------
+        x : Predicted value for the time series at chosen timestep (time). 
+        """
+        v = list(ts.values())
+        x = sm.tsa.ARMA(v, (2,0)).fit(disp=-1).forecast(time)[0][time-1]
+        return x
+
+    
+    def predict_arch(ts, time):
+        """
+        Predict the value of supply or demand at a given time step using the 
+        currently available time series data. This method impliments an ARCH
+        calculation to perform the prediciton. 
+        """
+
+    """This dictionary is used to select the calculation method used in the 
+    institution. 
+    """
+    calc_methods = {
```

Comment:
Global constants should be defined at the top of the file and should use `CAPS_WITH_UNDERSCORES` in their name